### ISSUE TITLE
代码理解问题

### ISSUE THREAD

**Issue by hunaghuohui** [Coder-Yu/SELFRec#27]
你好，请问下面XSimGCL模型中的这行代码怎么理解
ego_embeddings += torch.sign(ego_embeddings) * F.normalize(random_noise, dim=-1) * self.eps
这应该是论文中数据增强部分，是否对应sign（e）点乘X，如果是的话，X我没有理解，如果不是的话，请问 F.normalize(random_noise, dim=-1) * self.eps这部分是什么意思？

**Comment by hunaghuohui** [comment]
你好，F.normalize(random_noise, dim=-1) * self.eps，这个操作可以让点分布在球面上，但是这个和均匀分布有什么联系，我不太清楚
